drivers/clk/clk-hifiberry-dacpro.c

```diff
@@ -121,15 +123,34 @@ static int clk_hifiberry_dacpro_probe(struct platform_device *pdev)
 	ret = devm_clk_hw_register_clkdev(dev, &proclk->hw,
 					  init.name, NULL);
 #endif
+	if (ret) {
+		dev_err(dev, "Fail to add clock driver\n");
+		return ret;
+	}
+
+	proclk->sclk_lookup = clkdev_hw_create(&proclk->hw, "sclk", NULL);
+	if (!proclk->sclk_lookup) {
+#ifndef CONFIG_ACPI
+		of_clk_del_provider(dev->of_node);
+#endif
+		ret = -ENOMEM;
```

Comment:
we're failing the probe here, right? So should we really still call `platform_set_drvdata()` below? Wouldn't an immediate `return` here be more correct?

Reply:
you're right, will fix, thanks for the review

Reply:
@plbossart looks like this one got lost?

Reply:
no, it you can't get a lookup you return -ENOMEM. I don't understand the comment?

Reply:
@plbossart If lookup fails, r is set to -ENOMEM but we don't return from function here, but continue to L139 and call set_drvdata in error case as well. This seems wrong (although probably harmless as the probe will fail).